core/src/test/java/org/elasticsearch/rest/action/support/RestTableTests.java

```diff
@@ -70,6 +71,55 @@ public void testThatDisplayHeadersAreNotAddedTwice() throws Exception {
         assertThat(headerNames, not(hasItem("unmatched")));
     }
 
+    public void testThatWeUseTheAcceptHeaderJson() throws Exception {
+        assertBuildResponse(Collections.singletonMap("Accept", "application/json"), "application/json",
+                "[{\"bulk.foo\":\"foo\",\"bulk.bar\":\"foo\",\"aliasedBulk\":\"foo\"," +
+                        "\"aliasedSecondBulk\":\"foo\",\"unmatched\":\"foo\"," +
+                        "\"invalidAliasesBulk\":\"foo\"}]");
+    }
+
+    public void testThatWeUseTheAcceptHeaderText() throws Exception {
+        assertBuildResponse(Collections.singletonMap("Accept", "text/plain"), "text/plain", "foo foo foo foo foo foo \n");
+    }
+
+    public void testThatWeUseTheContentTypeHeaderJson() throws Exception {
+        assertBuildResponse(Collections.singletonMap("Content-Type", "application/json"), "application/json",
```

Comment:
In these cases its acceptable to use randomize testing's `rarely()` or its like to cover either branch randomly.